src/include/processor/processor_task.h

```diff
@@ -14,14 +14,18 @@ class ProcessorTask : public common::Task {
 
     void run() override;
     void finalizeIfNecessary() override;
+    void addSource(PhysicalOperator* source);
```

Comment:
instead of addSource, you might wanna just change the constructor

Reply:
I implemented this so I could update the progress bar in the `Task::deRegisterThreadAndFinalizeTask()` so this will be removed as it will be unused after I move the `updateProgress()` function